std/stdio.d

```diff
@@ -1767,9 +1767,30 @@ is recommended if you want to process a complete file.
      * Read data from the file according to the specified
      * $(LINK2 std_format.html#_format-string, _format specifier) using
      * $(REF formattedRead, std,_format).
+     * Example:
+----
+// test.d
+void main()
+{
+    import std.stdio;
+    auto f = File("input");
+    foreach (_; 0 .. 3)
+    {
+        int a;
+        f.readf(" %d", a);
+        writeln(++a);
+    }
+}
+----
+$(CONSOLE
+% echo "1 2 3" > input
+% rdmd test.d
+2
+3
+4
+)
      */
-    uint readf(Data...)(in char[] format, Data data)
-    if (allSatisfy!(isPointer, Data))
+    uint readf(Data...)(in char[] format, auto ref Data data)
```

Comment:
Yes that's the trick to be able to still accept pointers ;-)
AFAICT are pointers no lvalues in D.

We could add a constraint like (in pseudo syntax):
```d
All!(T => isRef!T || isPointer!T)
```